packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -576,6 +576,20 @@ class RunCommand extends RunCommandBase {
     webMode = featureFlags.isWebEnabled &&
       devices!.length == 1  &&
       await devices!.single.targetPlatform == TargetPlatform.web_javascript;
+
+    final String? flavor = stringArg('flavor');
+    if (flavor != null) {
+      const List<PlatformType> platformsWithFlavorsSupport = <PlatformType>[
+        PlatformType.android,
+        PlatformType.ios,
+        PlatformType.macos,
```

Comment:
Can each device type instead declare whether it supports flavors?

https://github.com/flutter/flutter/blob/995a020e48aa6627ea3f904a59dd56349a819eb4/packages/flutter_tools/lib/src/device.dart#L759-L773

Reply:
+1

Reply:
Great idea! Done.